coda/coda_mdstore/tests/test_views.py

```diff
@@ -0,0 +1,780 @@
+
+    def test_with_with_valid_coda_identifier_renders_xml(self, rf):
+        bag = FullBagFactory.create(name='coda-001')
+        ExternalIdentifierFactory.create(belong_to_bag=bag)
+
+        request = rf.get('/')
+        response = views.externalIdentifierSearch(request, bag.name)
+
+        assert response['Content-Type'] == 'application/atom+xml'
+        assert response.status_code == 200
+
+    def test_content_with_coda_identifier(self, rf):
+        bag = FullBagFactory.create(name='coda-001')
+        ExternalIdentifierFactory.create(belong_to_bag=bag)
+
+        request = rf.get('/')
+        response = views.externalIdentifierSearch(request, bag.name)
+
+        bagxml = objectify.fromstring(response.content)
+        bag_entry = bagxml.entry.content[self.CODA_XML]
+
+        assert str(bag_entry.bagitVersion) == bag.bagit_version
+        assert bag_entry.payloadSize == bag.size
+        assert bag_entry.fileCount == bag.files
+        assert bag_entry.name == bag.name
+        assert len(list(bag_entry.bagInfo.iterchildren())) == 2
+
+    def test_with_with_valid_metadc_identifier_renders_xml(self, rf):
+        bag = FullBagFactory.create()
+        ext_id = ExternalIdentifierFactory.create(
+            belong_to_bag=bag,
+            value='metadc000001'
+        )
+
+        request = rf.get('/')
+        response = views.externalIdentifierSearch(request, ext_id.value)
+
+        assert response['Content-Type'] == 'application/atom+xml'
+        assert response.status_code == 200
+
+    def test_content_with_metadc_identifier(self, rf):
+        bag = FullBagFactory.create()
+        ext_id = ExternalIdentifierFactory.create(
+            belong_to_bag=bag,
+            value='metadc000001'
+        )
+
+        request = rf.get('/')
+        response = views.externalIdentifierSearch(request, ext_id.value)
+
+        bagxml = objectify.fromstring(response.content)
+        bag_entry = bagxml.entry.content[self.CODA_XML]
+
+        assert str(bag_entry.bagitVersion) == bag.bagit_version
+        assert bag_entry.payloadSize == bag.size
+        assert bag_entry.fileCount == bag.files
+        assert bag_entry.name == bag.name
+        assert len(list(bag_entry.bagInfo.iterchildren())) == 2
+
+    def test_request_with_short_ark_parameter(self, client, rf):
+        bag = FullBagFactory.create()
+        ext_id = ExternalIdentifierFactory.create(
+            belong_to_bag=bag,
+            value='ark:/67531/metadc000001'
+        )
+
+        # URL with the identifier as a url paramater. We will feed the url to
+        # the request factory because the view will use the path to determine
+        # the id field.
+        url = reverse(
+            'coda_mdstore.views.externalIdentifierSearch', args=[ext_id.value])
+        request = rf.get(url)
+        response1 = views.externalIdentifierSearch(request, ext_id.value)
+
+        # URL with the ark id as a query paramater.
```

Comment:
Not looking for typos really at the moment, but: paramater --> parameter. Also, do you want to capitalize instance of "url" throughout?